satpy/etc/composites/abi.yaml

```diff
@@ -679,3 +679,22 @@ composites:
     prerequisites:
       - name: C14
     standard_name: highlighted_toa_brightness_temperature
+
+  # NOAA GOES-R Level-2 ABI Cloud Mask product
+  binary_cloud_mask:
+    compositor: !!python/name:satpy.composites.GenericCompositor
+    prerequisites:
+      - BCM
+    standard_name: binary_cloud_mask
+
+  four_level_cloud_mask:
+    compositor: !!python/name:satpy.composites.GenericCompositor
+    prerequisites:
+      - ACM
+    standard_name: four_level_cloud_mask
+
+  cloud_probability:
+    compositor: !!python/name:satpy.composites.GenericCompositor
+    prerequisites:
+      - Cloud_Probabilities
+    standard_name: cloud_probability
```

Comment:
Do we need separate composites for this? Could you make the enhancements use `name: BCM` (for example) and have this work the same?

Reply:
Or does the enhancement change the end result so much that you would prefer them to be separate products?

Reply:
These composites added customized color map/color palette to the visualization of these variables. In my experience these color maps are quite useful especially for the 4-level cloud mask and the cloud probability; because without the color map the default image will be in grey scale, which is hard to differentiate the different mask levels/cloud probabilities. 
The binary cloud mask (BCM) works OK in a black/white image, however, by assign same colors scheme as the new 4-level-cloud-mask, it is easy to see the difference/improvement of the new products.

Reply:
Sure, but the composite isn't doing anything here. Why not remove the composite definition and update the enhancements to use the product from the reader directly? That way someone doing `.load(['BCM'])` would get the nice colorized version by default.

Reply:
OK, I see what you mean now, I will try to make it as default in the enhancement config file.

Reply:
I have given a second though and think the composites are still necessary, because the cloud masks are not only use for visualization, but also can be used for masking other products. In the later case, user need to load the values in the data file instead of the colors by default enhancement; for example, to generate an ABI cloud product similar to the [one for SEVIRI](https://github.com/pytroll/satpy/blob/72dbcd25696e39423a56c078c3d7038e19bfdc7b/satpy/etc/composites/seviri.yaml#L29). Is that make sense?

Reply:
The enhancements are **only** applied when saving an image (PNG, geotiff, etc). They will not effect someone using the dataset inside a composite unless the composite explicitly pre-enhances the dataset. For example, the DayNightCompositor could put cloud mask on the day side and night microphysics on the night side of an image, but it must enhance both of them before combining them.

If someone wants to use a cloud mask outside of Satpy then they shouldn't be saving it to an image format and should instead either be using NetCDF or saving as a geotiff with `enhance=False` to make sure the enhancement is not used.

Let me know if I'm missing something here.

Reply:
As for your reference of the seviri cloud product, doesn't that `ct` that is used in that composite come directly from the reader?

Reply:
Yes, you are right. I realized that the enhancement is applied at the stage save dataset to image not at the time loading the dataset. I confused the functionalities which applied to the 'raw' dataset during the loading with enhancement at certain point. Thanks for clarification.